pkg/datapath/loader/netlink.go

```diff
@@ -97,12 +67,6 @@ func replaceDatapath(ctx context.Context, opts replaceDatapathOptions) (_ func()
 		}
 	}
 
-	for _, prog := range opts.programs {
-		if spec.Programs[prog.progName] == nil {
```

Comment:
We don't check for the presence of the programs anymore, at least add a nil check in attachXDP or similar?

Reply:
I know, but we don't pass a Collection into attachXDP/SKBProgram, so the nil check would have to be added to every call site. Realistically, these should never fail, but of course panic is never good. The plan is to replace this with LoadAndAssign in the very near future.

Having to pass a Collection to attach* is quite annoying for testing purposes.